tests/modules/programs/git-with-str-extra-config.nix

```diff
@@ -0,0 +1,22 @@
+{ config, lib, ... }:
+
+with lib;
+
+{
+  config = {
+    programs.git = {
+      enable = true;
+      extraConfig = ''
+        This can be anything.
+      '';
+      userEmail = "user@example.org";
+      userName = "John Doe";
+    };
+
+    nmt.script = ''
+      assertFileExists home-files/.config/git/config
+      assertFileContent home-files/.config/git/config \
```

Comment:
It would be nice to have something like `assertFileContains` that can be used to test that file contains some part of the configuration. Because right now this test will fail if we will change the order in which config sections are produced. I'm not sure how important it will be though.

Reply:
The order in files generated from attribute sets should be deterministic (since attribute sets are sorted) but yeah, more assertions could be added. There is one for matching based on regular expressions: [`assertFileRegex`](https://gitlab.com/rycee/nmt/blob/4d7b4bb34ed9df333b5aa54509e50881f3a59939/bash-lib/assertions.sh#L23-33). More can be added as need arise 🙂